api/sys_plugins_runtimes.go

```diff
@@ -0,0 +1,222 @@
+	ctx, cancelFunc := c.c.withConfiguredTimeout(ctx)
+	defer cancelFunc()
+
+	path := pluginRuntimeCatalogPathByType(i.Type, i.Name)
+	req := c.c.NewRequest(http.MethodGet, path)
+
+	resp, err := c.c.rawRequestWithContext(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var result struct {
+		Data *GetPluginRuntimeResponse
+	}
+	err = resp.DecodeJSON(&result)
+	if err != nil {
+		return nil, err
+	}
+	return result.Data, err
+}
+
+// RegisterPluginRuntimeInput is used as input to the RegisterPluginRuntime function.
+type RegisterPluginRuntimeInput struct {
+	// Name is the name of the plugin. Required.
+	Name string `json:"-"`
+
+	// Type of the plugin. Required.
+	Type PluginRuntimeType `json:"type"`
+
+	OCIRuntime   string  `json:"oci_runtime,omitempty"`
+	CgroupParent string  `json:"cgroup_parent,omitempty"`
+	CPU          float32 `json:"cpu,omitempty"`
+	Memory       uint64  `json:"memory,omitempty"`
+}
+
+// RegisterPluginRuntime wraps RegisterPluginWithContext using context.Background.
+func (c *Sys) RegisterPluginRuntime(i *RegisterPluginRuntimeInput) error {
+	return c.RegisterPluginRuntimeWithContext(context.Background(), i)
+}
+
+// RegisterPluginRuntimeWithContext registers the plugin with the given information.
+func (c *Sys) RegisterPluginRuntimeWithContext(ctx context.Context, i *RegisterPluginRuntimeInput) error {
+	ctx, cancelFunc := c.c.withConfiguredTimeout(ctx)
+	defer cancelFunc()
+
+	path := pluginRuntimeCatalogPathByType(i.Type, i.Name)
+	req := c.c.NewRequest(http.MethodPut, path)
+
+	if err := req.SetJSONBody(i); err != nil {
+		return err
+	}
+
+	resp, err := c.c.rawRequestWithContext(ctx, req)
+	if err == nil {
+		defer resp.Body.Close()
+	}
+	return err
+}
+
+// DeregisterPluginRuntimeInput is used as input to the DeregisterPluginRuntime function.
+type DeregisterPluginRuntimeInput struct {
+	// Name is the name of the plugin runtime. Required.
+	Name string `json:"-"`
+
+	// Type of the plugin. Required.
+	Type PluginRuntimeType `json:"type"`
+}
+
+// DeregisterPluginRuntime wraps DeregisterPluginRuntimeWithContext using context.Background.
+func (c *Sys) DeregisterPluginRuntime(i *DeregisterPluginRuntimeInput) error {
+	return c.DeregisterPluginRuntimeWithContext(context.Background(), i)
+}
+
+// DeregisterPluginRuntimeWithContext removes the plugin with the given name from the plugin
+// catalog.
+func (c *Sys) DeregisterPluginRuntimeWithContext(ctx context.Context, i *DeregisterPluginRuntimeInput) error {
+	ctx, cancelFunc := c.c.withConfiguredTimeout(ctx)
+	defer cancelFunc()
+
+	path := pluginRuntimeCatalogPathByType(i.Type, i.Name)
+	req := c.c.NewRequest(http.MethodDelete, path)
+	resp, err := c.c.rawRequestWithContext(ctx, req)
+	if err == nil {
+		defer resp.Body.Close()
+	}
+	return err
+}
+
+// ListPluginRuntimes wraps ListPluginRuntimesWithContext using context.Background.
+func (c *Sys) ListPluginRuntimes(i *ListPluginRuntimesInput) (*ListPluginRuntimesResponse, error) {
```

Comment:
The `X` and `XWithContext` pattern of functions only exists for backwards compatibility reasons. No need to implement context-less functions for new API surface IMO, just implement `ListPluginRuntimes` and make the first argument a context.